File: src/main/java/net/softbell/bsh/controller/rest/api/v1/TriggerRestV1.java
package net.softbell.bsh.controller.rest.api.v1;

import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import lombok.RequiredArgsConstructor;
import net.softbell.bsh.domain.entity.NodeTrigger;
import net.softbell.bsh.dto.response.ResultDto;
import net.softbell.bsh.iot.service.v1.IotTriggerServiceV1;
import net.softbell.bsh.service.ResponseService;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 트리거 관련 REST API 컨트롤러 V1
 */
@RequiredArgsConstructor
@RestController
@RequestMapping("/api/rest/v1/trigger")
public class TriggerRestV1
{
	// Global Field
	private final ResponseService responseService;
    private final IotTriggerServiceV1 iotTriggerService;
	
	@PostMapping("/status/{id}")
	public ResultDto setTriggerStatus(Authentication auth, @PathVariable("id")long id, @RequestParam("status")boolean status)
	{
		// Field
		boolean isSuccess;
		
		// Init
		isSuccess = iotTriggerService.setTriggerEnableStatus(auth, id, status);
		
		// Return
		if (isSuccess)
			return responseService.getSuccessResult();
		else
			return responseService.getFailResult(-10, "해당하는 아이템이 없음");
	}
}

File: src/main/kotlin/net/softbell/bsh/controller/rest/api/v1/TriggerRestV1.kt
package net.softbell.bsh.controller.rest.api.v1

import net.softbell.bsh.dto.response.ResultDto
import net.softbell.bsh.iot.service.v1.IotTriggerServiceV1
import net.softbell.bsh.service.ResponseService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.Authentication
import org.springframework.web.bind.annotation.*

/**
 * @author : Bell(bell@softbell.net)
 * @description : 트리거 관련 REST API 컨트롤러 V1
 */
@RestController
@RequestMapping("/api/rest/v1/trigger")
class TriggerRestV1 {
    // Global Field
    @Autowired private lateinit var responseService: ResponseService
    @Autowired private lateinit var iotTriggerService: IotTriggerServiceV1

    @PostMapping("/status/{id}")
    fun setTriggerStatus(auth: Authentication, @PathVariable("id") id: Long, @RequestParam("status") status: Boolean): ResultDto {
        // Init
        val isSuccess = iotTriggerService.setTriggerEnableStatus(auth, id, status)

        // Return
        return if (isSuccess)
            responseService.getSuccessResult()
        else
            responseService.getFailResult(-10, "해당하는 아이템이 없음")
    }
}